File: Knjigoteka.Model/Entities/SaleItem.cs
using Microsoft.EntityFrameworkCore;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Knjigoteka.Model.Entities
{
    public class SaleItem
    {
        [Key]
        public int Id { get; set; }
        [Required]
        public int SaleId { get; set; }
        [Required]
        public int BookId { get; set; }
        [Required]
        [Range(0, int.MaxValue, ErrorMessage = "Quantity must be greater than 0")]
        public int Quantity { get; set; }
        [Required]
        [Precision(18, 2)]
        public decimal UnitPrice { get; set; }
        [ForeignKey(nameof(BookId))]
        public Book Book { get; set; } = null!;
        [ForeignKey(nameof(SaleId))]
        public Sale Sale { get; set; } = null!;
    }

}

File: Knjigoteka.UI/knjigoteka_desktop/lib/models/sale_item.dart
class SaleItem {
  final int id;
  final int bookId;
  final String bookTitle;
  final int quantity;
  final double unitPrice;
  SaleItem({
    required this.id,
    required this.bookId,
    required this.bookTitle,
    required this.quantity,
    required this.unitPrice,
  });
  factory SaleItem.fromJson(Map<String, dynamic> json) => SaleItem(
    id: json['id'],
    bookId: json['bookId'],
    bookTitle: json['bookTitle'] ?? "",
    quantity: json['quantity'],
    unitPrice: (json['unitPrice'] as num).toDouble(),
  );
}
